java/serve/src/main/java/io/ray/serve/handle/DeploymentHandle.java

```diff
@@ -0,0 +1,128 @@
+
+  private transient volatile Router router;
+
+  public DeploymentHandle(
+      String deploymentName, String appName, HandleOptions handleOptions, Router router) {
+    this.deploymentId = new DeploymentId(deploymentName, appName);
+    this.handleOptions = handleOptions != null ? handleOptions : new HandleOptions();
+    this.handleTag =
+        StringUtils.isBlank(appName)
+            ? deploymentName + "#" + RandomStringUtils.randomAlphabetic(6)
+            : appName + "#" + deploymentName + "#" + RandomStringUtils.randomAlphabetic(6);
+    this.router = router;
+    initMetrics();
+  }
+
+  private void initMetrics() {
+    Map<String, String> metricsTags = new HashMap<>();
+    metricsTags.put(RayServeMetrics.TAG_HANDLE, handleTag);
+    metricsTags.put(RayServeMetrics.TAG_ENDPOINT, deploymentId.getName());
+    if (StringUtils.isNotBlank(deploymentId.getApp())) {
+      metricsTags.put(RayServeMetrics.TAG_APPLICATION, deploymentId.getApp());
+    }
+    RayServeMetrics.execute(
+        () ->
+            this.requestCounter =
+                Metrics.count()
+                    .name(RayServeMetrics.SERVE_HANDLE_REQUEST_COUNTER.name())
+                    .description(RayServeMetrics.SERVE_HANDLE_REQUEST_COUNTER.getDescription())
+                    .unit("")
+                    .tags(metricsTags)
+                    .register());
+  }
+
+  /**
+   * Returns a Ray ObjectRef whose results can be waited for or retrieved using ray.wait or Ray.get
+   * (or ``await object_ref``), respectively.
+   *
+   * @param parameters The input parameters of the specified method to be invoked in the deployment.
+   * @return ObjectRef
+   */
+  public DeploymentResponse remote(Object... parameters) {
+    RayServeMetrics.execute(() -> requestCounter.inc(1.0));
+    RequestMetadata.Builder requestMetadata = RequestMetadata.newBuilder();
+    requestMetadata.setRequestId(RandomStringUtils.randomAlphabetic(10));
+    requestMetadata.setEndpoint(deploymentId.getName());
+    requestMetadata.setCallMethod(
+        handleOptions != null ? handleOptions.getMethodName() : Constants.CALL_METHOD);
+    return new DeploymentResponse(
+        getOrCreateRouter().assignRequest(requestMetadata.build(), parameters));
+  }
+
+  private Router getOrCreateRouter() {
+    if (router != null) {
+      return router;
+    }
+
+    synchronized (DeploymentHandle.class) {
+      if (router != null) {
+        return router;
+      }
+      router = new Router(Serve.getGlobalClient().getController(), deploymentId.getName());
+    }
+    return router;
+  }
+
+  /**
+   * When `DeploymentHandle` is passed as a parameter, `Router` and `Count` cannot be serialized, so
+   * we do not serialize them. During deserialization, we reconstruct them.
+   *
+   * @param in
+   * @throws IOException
+   * @throws ClassNotFoundException
+   */
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+    in.defaultReadObject();
```

Comment:
For my knowledge, will this automatically filter all non-serializable object in the class?

Reply:
No, this does not automatically filter out objects that are not serializable, so I have marked objects that are not serializable and not necessary to be serialized with the 'transient' modifier. Properties marked as 'transient' will be ignored during serialization.